stacks-node/src/burnchains/rpc/bitcoin_rpc_client/mod.rs

```diff
@@ -0,0 +1,510 @@
+                }),
+            ],
+        )?)
+    }
+
+    /// Mines a specified number of blocks and sends the block rewards to a given address.
+    ///
+    /// # Arguments
+    /// * `num_block` - The number of blocks to mine.
+    /// * `address` - The Bitcoin address to receive the block rewards.
+    ///
+    /// # Returns
+    /// A vector of block hashes corresponding to the newly generated blocks.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.17.0**.
+    ///
+    /// # Notes
+    /// Typically used on `regtest` or test networks.
+    pub fn generate_to_address(
+        &self,
+        num_block: u64,
+        address: &str,
+    ) -> BitcoinRpcClientResult<Vec<String>> {
+        Ok(self.global_ep.send(
+            &self.client_id,
+            "generatetoaddress",
+            vec![num_block.into(), address.into()],
+        )?)
+    }
+
+    /// Retrieves detailed information about an in-wallet transaction.
+    ///
+    /// This method returns information such as amount, fee, confirmations, block hash,
+    /// hex-encoded transaction, and other metadata for a transaction tracked by the wallet.
+    ///
+    /// # Arguments
+    /// * `txid` - The transaction ID (txid) to query, as a hex-encoded string.
+    ///
+    /// # Returns
+    /// A [`GetTransactionResponse`] containing detailed metadata for the specified transaction.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.10.0**.
+    pub fn get_transaction(&self, txid: &str) -> BitcoinRpcClientResult<GetTransactionResponse> {
+        Ok(self
+            .wallet_ep
+            .send(&self.client_id, "gettransaction", vec![txid.into()])?)
+    }
+
+    /// Broadcasts a raw transaction to the Bitcoin network.
+    ///
+    /// This method sends a hex-encoded raw Bitcoin transaction. It supports optional limits for the
+    /// maximum fee rate and maximum burn amount to prevent accidental overspending.
+    ///
+    /// # Arguments
+    ///
+    /// * `tx` - A hex-encoded string representing the raw transaction.
+    /// * `max_fee_rate` - Optional maximum fee rate (in BTC/kvB). If `None`, defaults to `0.10` BTC/kvB.
+    ///     - Bitcoin Core will reject transactions exceeding this rate unless explicitly overridden.
+    ///     - Set to `0.0` to disable fee rate limiting entirely.
+    /// * `max_burn_amount` - Optional maximum amount (in satoshis) that can be "burned" in the transaction.
+    ///     - Introduced in Bitcoin Core v25 (https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-25.0.md#rpc-and-other-apis)
+    ///     - If `None`, defaults to `0`, meaning burning is not allowed.
+    ///
+    /// # Returns
+    /// A transaction ID as a `String`.
+    ///
+    /// # Availability
+    /// - **Since**: Bitcoin Core **v0.7.0**.
+    /// - `maxburnamount` parameter is available starting from **v25.0**.
+    pub fn send_raw_transaction(
+        &self,
+        tx: &str,
```

Comment:
Please use `stackslib::burnchains::bitcoin::BitcoinTransaction` instead of `&str`.

Reply:
Thanks for catching this!  You're right -- `BitcoinTransaction` does not offer a way to convert itself into a byte array.  Let's either continue to use `SerializedTx`, or just take a `Transaction` and have `send_transaction()` use code from `SerializedTx::new()` to convert it into a byte string.